src/util_libc.rs

```diff
@@ -36,3 +36,29 @@ impl Weak {
         NonNull::new(addr as *mut _)
     }
 }
+
+pub struct LazyFd(LazyUsize);
+
+impl LazyFd {
+    pub const fn new() -> Self {
+        Self(LazyUsize::new())
+    }
+
+    // If init() returns Some(x), x should be nonnegative.
+    pub fn init(&self, init: impl FnOnce() -> Option<libc::c_int>) -> Option<libc::c_int> {
+        let fd = self.0.sync_init(
+            || match init() {
+                // OK as val >= 0 and val <= c_int::MAX < usize::MAX
+                Some(val) => val as usize,
+                None => LazyUsize::UNINIT,
+            },
+            || unsafe {
+                libc::usleep(1000);
```

Comment:
Wow that's a long sleep, isn't it? A comment seems always warranted with such a magic number.

Reply:
Ya this sleep should probably be shorter, any recommendations? My guess is that we would want it to be longer than the normal latency for an `open(2)` syscall.

Reply:
On my laptop `libc::open` takes ~7µs and on playground ~8µs, so I think 10µs will be a good value.

Reply:
I'm sorry this is outside my range of experience.^^ 1ms just seemed like a long time. `/dev/random` doesn't even do "real" I/O, so it shouldn't have *huge* latency.

Reply:
It can be a huge latency if system entropy pool has not initialized, IIRC up to several minutes.

Reply:
Sure, but that's not the common case.

Ideally `wait` would do something like double the wait time each time around the loop, starting at e.g. 1µs and maxing out at 1s or so. But that seems like overkill.^^